pkg/network/ebpf/c/shared-libraries/probes.h

```diff
@@ -82,12 +82,23 @@ static __always_inline void do_sys_open_helper_exit(exit_sys_ctx *args) {
         goto cleanup;
     }
 
-    if (!match6chars(0, 'l', 'i', 'b', 's', 's', 'l') && !match6chars(0, 'c', 'r', 'y', 'p', 't', 'o') && !match6chars(0, 'g', 'n', 'u', 't', 'l', 's') && !match6chars(0, 'c', 'u', 'd', 'a', 'r', 't')) {
-        goto cleanup;
+    u64 crypto_libset_enabled = 0;
+    LOAD_CONSTANT("crypto_libset_enabled", crypto_libset_enabled);
+
+    if (crypto_libset_enabled && (match6chars(0, 'l', 'i', 'b', 's', 's', 'l') || match6chars(0, 'c', 'r', 'y', 'p', 't', 'o') || match6chars(0, 'g', 'n', 'u', 't', 'l', 's'))) {
+        // Only get the CPU if we're going to use it.
+        u32 cpu = bpf_get_smp_processor_id();
+        bpf_perf_event_output((void *)args, &crypto_shared_libraries, cpu, path, sizeof(lib_path_t));
+    }
```

Comment:
I'd add here "goto cleanup"
As if we matched on crypto libraries, we will not match on the gpu libraries

Reply:
Fixed